src/test/java/org/folio/util/UrlUtilTest.java

```diff
@@ -1,76 +1,83 @@
-import org.folio.util.model.UrlCheckResult;
-import org.junit.After;
+import java.io.IOException;
+import java.net.MalformedURLException;
+
+import org.folio.rest.tools.utils.NetworkUtils;
+import org.junit.AfterClass;
 import org.junit.Before;
-import org.junit.Rule;
+import org.junit.BeforeClass;
 import org.junit.Test;
 import org.junit.runner.RunWith;
 import org.slf4j.Logger;
 import org.slf4j.LoggerFactory;
 
-import java.io.IOException;
-import java.net.MalformedURLException;
-import java.net.ServerSocket;
+import io.vertx.core.DeploymentOptions;
+import io.vertx.core.Vertx;
+import io.vertx.core.json.JsonObject;
+import io.vertx.ext.unit.TestContext;
+import io.vertx.ext.unit.junit.VertxUnitRunner;
 
 @RunWith(VertxUnitRunner.class)
 public class UrlUtilTest {
 
   private static final Logger log = LoggerFactory.getLogger(UrlUtilTest.class);
 
-  @Rule
-  public RunTestOnContext rule = new RunTestOnContext();
+  public static final int MOCK_PORT = NetworkUtils.nextFreePort();
 
-  @Rule
-  public RepeatRule repeatRule = new RepeatRule();
+  private static Vertx mockVertx = Vertx.vertx();
 
-  private HttpServer server;
+  private Vertx vertx;
 
+  @BeforeClass
+  public static void setupOnce(TestContext context) throws Exception {
+    DeploymentOptions mockOptions = new DeploymentOptions().setConfig(new JsonObject()
+        .put("http.port", MOCK_PORT))
+        .setWorker(true);
+
+    mockVertx.deployVerticle(IdpMock.class.getName(), mockOptions, context.asyncAssertSuccess());
+  }
 
   @Before
   public void before(TestContext context) throws IOException {
+    vertx = Vertx.vertx();
+    WebClientFactory.init(vertx);
+  }
 
-    // obtain a free port
-    ServerSocket ss = new ServerSocket(0);
-    int port = ss.getLocalPort();
-    ss.close();
-
-    server = rule.vertx().createHttpServer().requestHandler(req -> {
-      req.response()
-        .putHeader("Content-Type", "application/xml")
-        .end("<?xml version=\"1.0\" encoding=\"UTF-8\"?>");
-    }).
-      listen(port, context.asyncAssertSuccess(hnd -> {
-        log.info("Running test http listener on port " + hnd.actualPort());
-      }));
+  @AfterClass
+  public static void afterOnce(TestContext context) {
+    mockVertx.close(context.asyncAssertSuccess());
+  }
 
+  @Test
+  public void checkIdpUrl(TestContext context) throws MalformedURLException {
+    UrlUtil.checkIdpUrl("http://localhost:" + MOCK_PORT + "/xml", vertx)
+      .onComplete(context.asyncAssertSuccess(result -> {
+        context.assertEquals("", result.getMessage());
+      }));
   }
 
-  @After
-  public void after(TestContext context) {
-    server.close(context.asyncAssertSuccess());
+  @Test
+  public void checkIdpUrlNon200(TestContext context) {
+    UrlUtil.checkIdpUrl("http://localhost:0", vertx)
+      .onComplete(context.asyncAssertSuccess(result -> {
+          context.assertEquals("Connection refused: localhost/127.0.0.1:0", result.getMessage());
```

Comment:
This test failed on my Mac with error `java.lang.AssertionError: Not equals : Connection refused: localhost/127.0.0.1:0 != Can't assign requested address: localhost/127.0.0.1:0`. Maybe it should call `NetworkUtils.nextFreePort()` to get a port rather than hardcode port `0` here.